r1cs/gadgets/std/src/groups/mod.rs

```diff
@@ -775,28 +830,51 @@ pub(crate) mod test {
         ConstraintF: Field,
         G: EndoMulCurve,
         GG: EndoMulCurveGadget<G, ConstraintF, Value = G>,
-    >() {
-        let mut cs = ConstraintSystem::<ConstraintF>::new(SynthesisMode::Debug);
-
-        let a_native = G::rand(&mut thread_rng());
-        let a = GG::alloc(&mut cs.ns(|| "generate_a"), || Ok(a_native)).unwrap();
-
-        let scalar: G::ScalarField = u128::rand(&mut thread_rng()).into();
-
-        let b_native = scalar.into_repr().to_bits().as_slice()[0..128].to_vec();
-        let b = b_native
-            .iter()
-            .map(|&bit| Boolean::constant(bit))
-            .collect::<Vec<_>>();
-
-        let r_native = a_native.endo_mul(b_native).unwrap();
-        let r = a
-            .endo_mul(cs.ns(|| "endo mul"), &b)
-            .unwrap()
-            .get_value()
-            .unwrap();
+    >() 
+    {
+        for _ in 0..10 {
+            let mut cs = ConstraintSystem::<ConstraintF>::new(SynthesisMode::Debug);
 
-        assert_eq!(r_native, r);
+            let a_native = G::rand(&mut thread_rng());
+            let a_g = GG::alloc(&mut cs.ns(|| "generate_a"), || Ok(a_native)).unwrap();
+    
+            let scalar: G::ScalarField = u128::rand(&mut thread_rng()).into();
+            let mut scalar_bits_native = scalar.write_bits();
+            scalar_bits_native.reverse();
+            scalar_bits_native = scalar_bits_native.as_slice()[..128].to_vec();
+            
+            assert!(scalar_bits_native.len() == 128);
+            assert!(!scalar_bits_native.iter().all(|b| !b));
```

Comment:
I think this is just a check that I truncated correctly (e.g. I took the non-zero half). For sure If I sample 0 this will return an error (with overwhelmingly low probability). I'm gonna fix it anyway